### DISCUSSION TITLE
Does the current default threshold work well for you?

### DISCUSSION THREAD

**Discussion by Breakthrough** [Breakthrough/DVR-Scan#74] [General]
Hello all;

I would like some feedback on the default sensitivity threshold used by DVR-Scan (currently 0.15).  Please post here if the default sensitivity threshold is meeting your needs, or if you think it is too high / too low.  Anyone willing to share their experience and ranges of threshold values they used.

For v2.0 I would like to normalize the sensitivity, as currently it is defined to be between 0 and 255.0.  The new definition would be how much of the frame must have motion as a percentage from 0.0% (any pixel) to 100.0% (all pixels). This means the current default would then map from 0.15 to 0.15 / 255, which is only 0.0588% of the entire frame.  Thus I'm wondering if that might be too small for most use cases, and the default should be raised a bit higher to something like 2 or 3 to avoid false positives.

Looking forward to hearing any feedback anyone has on the topic!

**Comment by ocram** [comment]
Yes, it does.

For situations where the complete frame is essentially static, and with the camera perhaps locking focus/exposure, i.e. for the average situation where you want to track *any* motion and have ensured decent quality, it works perfectly.

If you want to catch tiny bits of motion (in size, not duration), you may even have to decrease it to, say, 0.1.

Of course, if you have a tree with leaves moving in the wind in the frame, and for some reason you cannot apply a ROI, 0.15 (or even 0.1) will certainly not work for you.

Anyway, it’s important to consider that tuning with `--min-event-length` and `--kernel-size` in particular has a big impact as well. If your frame has constant disturbances of incredibly short duration, perhaps single-frame, or if there is too much noise, if you don’t adjust these two parameters accordingly, 0.15 will not work for you.

But yes, everything else taken account of, 0.15 seems like a good default for motion detection where the parts between motion are essentially static.

**Comment by ceewanna** [comment]
v1.5.1
My case threshold is something below 2.0. But my downscale-factor is set at 3 (my expectation is to process 1920x1080 based on 640x360.) I also had frame-skip set at 1 (my expectation is that this will not load much on cpu.)

Before I run into this topic, I had no idea what is the base of this threshold. The threshold was set way higher at 10 or 20. That was lowered to around 1.5 now. This is based on scene where there are a few people moving around.

**Comment by ceewanna** [comment]
After experimenting with CNT & MOG2, I found out that my rough tuning of threshold & kernel was invalid. Now I have a more detailed tuning of a real event video with results as follows.

Camera resolution 2560x1440
bg=CNT, all produce 3 events (1 false, 2 real events)
thr 0.15 kernel 17
thr 0.25 kernel 17
thr 0.5 kernel 13
thr 0.75 kernel 9 (still 3 detectable events)
thr 1.0 kernel --- (undetectable)
kernel default
thr 0.15-0.5 (1 false, 1st & 2nd events detected)
thr 0.75 (1 false, 1st event, miss 2nd)

bg=MOG2, produces 2 events (2 real events)
thr 0.15 kernel 21
thr 0.25 kernel 17
thr 0.5 kernel 13
thr 0.75 kernel 11 (1 real 1st event, miss the 2nd event)
thr 0.75 kernel 9 (3 events similar to CNT)
thr 1.0 kernel 9 (1 false (CNT) event, real 1st event, miss 2nd event)
thr 1.0 kernel 7 (1 false (CNT) event, real 1st event, miss 2nd event)
kernel default 
thr 0.25 (2 real events)
thr 0.5 (2 real events)
thr 0.75 (1st event detected, miss 2nd)

1st event : a bug creeping across camera
2nd event : a truck run across from left to right at around 30m distant
background green field with trees around

kernel shown above are the maximum achievable to detect the events.
The tests were all running with these options:
min-event-length = 1
downscale-factor = 2
frame-skip = 1
